src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
```

Comment:
PrimathonAICodeReviewer

The handling of the `err` variable seems inconsistent. It is checked for a `response` property, but then directly assigned the `response` property if it exists. Consider simplifying this logic for clarity.

```diff
- let err = exception['response'];
- err = err.response ? err.response : JSON.stringify(err);
+ const err = exception['response']?.response || JSON.stringify(exception['response']);
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->